### PR TITLE
Optimize member ptr funcs usage

### PR DESCRIPTION
**Pull request by vitamin-caig**
Hello!

Here's some optimizations I made to my fork of libsidplayfp long time ago.
Tl;dr, calling function via member to pointer may be quite expensive due to some additional checks compiler should perform.

Some performance reports. Rendering speed shows relation between rendered sound duration and spent real time (e.g. if 60s track is rendered in 10s, metric is 6.0)

Full tracks collection (68805 unique modules, 98188 tracks) rendered on Intel Xeon.

- Original: speed=10.02..62.29 avg=27.30 median=27.50
- Only CPU instructions optimized: speed=10.38..64.18 avg=28.12 median=28.28
- Also CPU events optimized: speed=10.45..66.33 avg=29.17 median=29.33

Optimization is 6.8% by average and 6.6% by median

(My fork with additional optimizations based on ancient version: speed=7.44..72.49 avg=31.53 median=30.34)

Random 1000 files from full collection (1413 tracks) rendered on Ryzen 5 1600.

- Original: speed=19.90..41.02 avg=33.72 median=34.00
- Only CPU instructions optimized: speed=21.06..43.59 avg=38.21 median=39.05
- Also CPU events optimized: speed=20.71..44.74 avg=39.14 median=40.03

Optimization is 16% by average and 17.7% by median

(My fork shows speed=10.80..55.75 avg=44.28 median=42.54)

Random 100 files from full collection (164 tracks) rendered on ARMv6 (RaspberryPi v1)

- Original: speed=1.06..1.97 avg=1.75 median=1.77
- Only CPU instructions optimized: speed=1.13..2.09 avg=1.83 median=1.85
- Also CPU events optimized: speed=1.04..2.07 avg=1.83 median=1.85

Optimization is 4.6% by average and 4.5% by median

(My fork shows speed=1.18..3.01 avg=2.47 median=2.51)

### CODE REVIEW

**Comment by drfiemost** [comment]
Nice improvement, thanks!
